torch/csrc/autograd/variable.h

```diff
@@ -333,40 +303,82 @@ struct Variable::Impl : public at::TensorImpl {
 
   std::shared_ptr<Function> get_grad_accumulator();
   virtual std::shared_ptr<Function>& get_grad_fn() {
-    return grad_fn;
+    return grad_fn_;
+  }
+
+  virtual const Variable& base() const {
+    throw std::runtime_error("Can't get base of non-view Variable");
+  }
```

Comment:
Why do we need this in `VariableImpl` now?

Reply:
I need to call `base()` from `requires_grad()`

Reply:
Can't we access the pointer directly?